cve_bin_tool/merge.py

```diff
@@ -44,9 +44,9 @@ def __init__(
     ):
         self.logger = logger or LOGGER.getChild(self.__class__.__name__)
         self.merge_files = merge_files
-        self.intermediate_cve_data = []
-        self.all_cve_data = []
-        self.file_stack = []
+        self.intermediate_cve_data: list = []
+        self.all_cve_data: list = []
+        self.file_stack: list = []
```

Comment:
cve-bin-tool supports Python 3.7-3.10, typing as `list` is only available since Python 3.9. You will need to import `List` from `typing` module and use it instead.
Moreover, these type hints are incomplete, you need to specify the type of list elements as well (`list[int]`, `list[str]` or whatever the type is).

Reply:
Changed as requested. In self.intermediate_cve_data, a JSON file is being read. So I had to use Any in that case

Reply:
Hm, I don't really like `Any`, I would look into the expected structure being read from JSON and try to define it but I'll let maintainers decide.

Reply:
What about `file_stack`? This can be `List[Union[str, Path]]`, it's better than adding more ambiguity with `Any`.
It's the easiest one at first glance but a better solution would probably be to change the logic to make it `Path` in all code paths.

Reply:
```py
if intermediate_path.is_dir():
    for filepath in self.walker([intermediate_path]):
        self.file_stack.append(filepath)
        yield filepath
        self.file_stack.pop()
elif intermediate_path.is_file() and not intermediate_path.is_symlink():
    self.file_stack.append(intermediate_path)
    yield intermediate_path
    self.file_stack.pop()
```

You are correct about the `if` branch but in `else` branch we append `intermediate_path` and it has `Path` type.

Reply:
So, should I typecast str to path in DirWalk class and make it return Iterator[Path]?